### ISSUE TITLE
if-branch context for version

### ISSUE THREAD

**Issue by tonyhffong** [tonyhffong/Lint.jl#58]
Reminder to add branch context for if statement so we don't get too much noises, such as

``` julia
if VERSION < v"0.3-"
   # lint should reference this constraints so we don't get false positives due to version differences
end
```
